### PR TITLE
Fix 1000 warnings

### PR DESCRIPTION
**Pull request by Memphiz**
What the topic says - added missing "override" keywords.

### CODE REVIEW

**Comment by Memphiz** [comment]
jenkins build and merge

**Comment by FernetMenta** [comment]
don't merge. I want to look over the interfaces in core in a controlled manner.

**Comment by Memphiz** [comment]
sure ... someone stopped the jenkins job in time already ...

**Comment by ksooo** [comment]
To whom it may concern: I'm currently patching my local xcode project file, because these warnings are so annoying: -Wno-inconsistent-missing-override ;-)

File: xbmc/windows/GUIMediaWindow.h [hunk 1]
```diff
@@ -38,19 +38,19 @@ class CGUIMediaWindow : public CGUIWindow
...
-  int GetViewCount() const { return m_viewControl.GetViewModeCount(); };
-  virtual bool HasListItems() const { return true; }
-  virtual CFileItemPtr GetCurrentListItem(int offset = 0);
+  virtual void OnWindowLoaded() override;
+  virtual void OnWindowUnload() override;
+  virtual void OnInitWindow() override;
+  virtual bool IsMediaWindow() const  override { return true; }
+  int GetViewContainerID() const  override{ return m_viewControl.GetCurrentControl(); }
+  int GetViewCount() const  override{ return m_viewControl.GetViewModeCount(); };
+  virtual bool HasListItems() const  override{ return true; }
```
Comment (Montellese) [inline xbmc/windows/GUIMediaWindow.h]: These three lines are missing a space betwen `override` and the following `{`.

**Comment by Memphiz** [comment]
fixed the missing spaces

**Comment by FernetMenta** [comment]
@ksooo even more annoying is incorrect behaviour. Much better than silencing warnings for the sake of silencing warnings is to properly review the interface. I use git blame quite often and don't want to lose history.

**Comment by ksooo** [comment]
> even more annoying is incorrect behaviour.

@FernetMenta sure. My comment was just a hint how to get a free view on "other" warnings until you have reviewed/fixed the core interfaces. Currently one can hardly see the few other warnings (one might have just introduced while working on new stuff locally).

**Comment by Paxxi** [comment]
@Memphiz you should remove virtual as override implies virtual making it redundant and of no value. Resharper/CLion will display it as dead code

**Comment by garbear** [comment]
@Paxxi I disagree, when assessing a function as virtual it is easier to identify the first word on the line as opposed to randomly placed within the line.

**Comment by Memphiz** [comment]
this pr is about compiler warnings and not codestyle

**Comment by Memphiz** [comment]
@FernetMenta if you have anything special to review (and want me to do it) in regard to this pr please let me know ...

**Comment by FernetMenta** [comment]
@Memphiz as already mentioned, I don't want sacrifice history this way. please exclude cores from this pr

**Comment by Memphiz** [comment]
excluded cores (which wasn't obvious to me that you wanted that...) - jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by Memphiz** [comment]
whats up with this *grr* jenkins build this please

**Pull request merged**